src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java

```diff
@@ -86,12 +87,32 @@ private long getAgeInMilliSeconds(Date birthday) {
     }
 
     private int getAge(Date birthday) {
-        Calendar c = Calendar.getInstance();
-        c.setTime(new Date());
-        int currentYear = c.get(Calendar.YEAR);
-        c.setTime(birthday);
-        int birthYear = c.get(Calendar.YEAR);
-        return currentYear - birthYear;
+        int years;
+        int months;
+        Calendar birthDayMili = Calendar.getInstance();
+        birthDayMili.setTimeInMillis(birthday.getTime());
+        long currentTime = System.currentTimeMillis();
+        Calendar now = Calendar.getInstance();
+        now.setTimeInMillis(currentTime);
+        years = now.get(Calendar.YEAR) - birthDayMili.get(Calendar.YEAR);
+        int currMonth = now.get(Calendar.MONTH) + 1;
+        int birthMonth = birthDayMili.get(Calendar.MONTH) + 1;
+        months = currMonth - birthMonth;
```

Comment:
This is funny, why do you add 1 to currMonth and birthMonth if you subtract one from another?

Reply:
fixed